test/modelnode.jl

```diff
@@ -338,3 +338,34 @@ end
         end
     end
 end
+
+gradients_match(grads1::Flux.Zygote.Grads, grads2::Flux.Zygote.Grads) = grads1.params == grads2.params && gradients_match(grads1.grads, grads2.grads)
+gradients_match(grads1::Flux.Zygote.Grads, grads2::Flux.Zygote.Grads) = grads1.params == grads2.params && gradients_match(grads1.grads, grads2.grads)
+gradients_match(grads1::IdDict, grads2::IdDict) = length(grads1) == length(grads2) && all(kv1 == kv2 for (kv1, kv2) in zip(grads1, grads2))
+
+@testset "testing simple named tuple model with reduce catobs in gradient" begin
+    layerbuilder(k) = Flux.Dense(k, 2, NNlib.relu)
+    x = ProductNode((node1 = BagNode(ArrayNode(randn(Float32, 3, 4)), [1:2, 3:4]),
+                     node2 = BagNode(ArrayNode(randn(Float32, 4, 4)), [1:1, 2:4])))
+    m = reflectinmodel(x, layerbuilder)
+    @test gradient(() -> sum(m[:node1]([x[:node1], x[:node1]]).data), params(m)) isa Flux.Zygote.Grads
```

Comment:
why not just `@test gradient(() -> sum(m([x, x]).data), params(m)) isa Flux.Zygote.Grads`? How is it different than the two following lines?

Reply:
ah, you're right, it can be simplified like this